### ISSUE TITLE
🐛 BUG: Astro.resolve() for sourcing scripts fails to get built, with and without hoist (astro@0.21.0-next.5)

### ISSUE THREAD

**Issue by bobaekang** [snowpackjs/astro#1851]
### What version of `astro` are you using?

0.21.0-next.5

### What package manager are you using?

npm

### What operating system are you using?

StackBlitz for the example, Windows (WSL 1) for my local use

### Describe the Bug

First of all, a huge thank you to Astro maintainers for tirelessly working on improving the project. I've been creating Issues without saying much on that, so I'd like to clarify that I'm very grateful for your contributions to pushing forward the web dev with Astro.

\* \* \*

This Issue is sort of a follow-up to #1715 and #1821.

The title pretty much says it all. Here's an example source:

```astro
---
---
<html lang="en">

<head>
	<meta charset="utf-8" />
	<link rel="icon" type="image/x-icon" href="/favicon.ico" />
	<meta name="viewport" content="width=device-width" />

	<!-- THIS DOESN'T WORK -->
	<script src={Astro.resolve('../script.js')} />
	
	<!-- THIS DOESN'T WORK EITHER -->
	<script src={Astro.resolve('../script-hoist.js')} hoist />

	<title>Welcome to Astro</title>
</head>

<body>
	<h1>Welcome to <a href="https://astro.build/">Astro</a></h1>
</body>

</html>
```

And here's the build output (note the `src` values):

```html
<!DOCTYPE html><html lang="en"><head>
	<meta charset="utf-8">
	<link rel="icon" type="image/x-icon" href="/favicon.ico">
	<meta name="viewport" content="width=device-width">

	<!-- THIS DOESN'T WORK -->
	<script src="/home/projects/github-4sqhr1-uhgrfh/src/script.js"></script>
	
	<!-- THIS DOESN'T WORK EITHER -->
	

	<title>Welcome to Astro</title>
<script src="/home/projects/github-4sqhr1-uhgrfh/src/script-hoist.js" hoist="true" astro-script="//script-0"></script></head>

<body>
	<h1>Welcome to <a href="https://astro.build/">Astro</a></h1>
</body></html>
```

The sourced scripts are not built at all as you can see in the `/dist` output:

![image](https://user-images.githubusercontent.com/22449454/142090935-51e50728-9a83-4d4e-9adf-5b79521e4898.png)

I have also tried the same thing on my local machine and observed the same result.[^1] 

[^1]: Curiously, the StackBlitz build somehow ignores `public/favicon.ico`, but it's probably their quirk since `/dist` on my machine includes `favicon.ico`.

In contrast, `Astro.resolve()` works with dev server.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-4sqhr1-uhgrfh

**Comment by jonathantneal** [comment]
Confirmed the unexpected output in `dev`:

```jsx
<!-- input -->
<script src={Astro.resolve(PATH)} hoist />
```

```html
<!-- output -->
<script src="RESOLVED_PATH" hoist="true" astro-script="//script-0"></script>
```

Confirmed the unexpected output and wrong path in `build`:

```html
<!-- output -->
<script src="/filesystem/path/to/project/src/script-hoist.js" hoist="true" astro-script="//script-0"></script>
```

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.

**Comment by bobaekang** [comment]
@FredKSchott Congratulations on v0.21 release!

Unfortunately, I can still reproduce the reported issue with `astro@0.21.0`--both on StackBlitz (https://stackblitz.com/edit/github-4sqhr1-v6c7qa; it's a fork of the link above) and my local machine (WSL1-Ubuntu 20.04). I haven't changed anything other than upgrading Astro to v0.21 and got the same results.
